### ISSUE TITLE
bug: broken support for yarn executing tests

### ISSUE THREAD

**Issue by jeripeierSBB** [ionic-team/stencil#3482]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/.github/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

2.17.1

### Current Behavior

Executing `yarn test` with another script name as `test` fails. 

Consider the following example specified in package.json scripts section:
```
...
 "scripts": {
    "test:dev": "stencil test --spec --e2e --coverage",
  },
...
```
if executing `yarn test:dev` it fails and the following output is generated:
```
jest args: test:dev --e2e --spec --coverage --max-workers=8
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/*/Desktop/test-fail-repro
  79 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: /Users/*/Desktop/test-fail-repro/.vscode, /Users/*/Desktop/test-fail-repro/.stencil, /Users/*/Desktop/test-fail-repro/node_modules, /Users/*/Desktop/test-fail-repro/www, /Users/*/Desktop/test-fail-repro/dist - 22 matches
  testRegex: (/__tests__/.*|(\.|/)(test|spec|e2e))\.[jt]sx?$ - 3 matches
Pattern: test:dev - 0 matches
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Expected Behavior

Tests should be executed as before version 2.17.1

### Steps to Reproduce

1. `npm init stencil`
2. move to your directory
3. execute `yarn`
4. rename test to `test:dev` in `package.json`
5. execute`yarn test:dev`
6. See error

### Code Reproduction URL

https://github.com/jeripeierSBB/stencil-yarn-fail-repro

**Comment by alicewriteswrongs** [comment]
Hello @jeripeierSBB, thanks for filing this issue! I was just able to validate that this is occurring. It look to be an issue specific to yarn, since in your reproduction case I am able to run the `test:dev` script with no problems w/ `npm` (i.e. `npm run test:dev`).

In any case, I'm going to add this to the backlog and hopefully we'll have a fix soon for running scripts like these w/ `yarn`!

**Comment by jeripeierSBB** [comment]
@alicewriteswrongs thanks for getting into this. I have to mention that with version 2.17.0 it worked, so this problem was newly introduced with version 2.17.1

**Comment by alicewriteswrongs** [comment]
That makes sense, this recent pr #3444 was [included in the 2.17.1 release](https://github.com/ionic-team/stencil/releases/tag/v2.17.1), so it makes sense that the behavior changed then.

**Comment by alicewriteswrongs** [comment]
If you're up for it, there's a bugfix PR up here: https://github.com/ionic-team/stencil/pull/3483

It will likely merge soon, but if you want to test it out to make sure it fixes your issue you could

- clone the repo locally
- check out [the branch](https://github.com/ionic-team/stencil/tree/ap/flag-parsing-bug)
- do `npm install`, `npm run build` and `npm pack`
- try installing the resulting `.tgz` file in our project by doing `npm install ./path/to/stencil.tgz`

If you'd like to wait for a pre-release or release build to try that out no worries :)

**Comment by alicewriteswrongs** [comment]
@jeripeierSBB would you mind providing some information about the version of yarn you're using, and how you had this working before upgrading to `@stencil/core@2.17.1`? I ask because in the reproduction repo that you provided I'm able to reproduce the issue with an earlier version of Stencil installed (`2.13.0` to be specific). I'm using yarn 1.22.18 for testing locally.

<img width="1030" alt="Screen Shot 2022-07-20 at 3 16 41 PM" src="https://user-images.githubusercontent.com/6207644/180064039-cc755d80-f5c3-4202-8f9d-356d2d5e7ced.png">

**Comment by manojreddymettu-pro** [comment]
Hi, This issue is happening with **NPM** as well.

Worked fine till `stencil/core:2.17.0` and after upgrading to `2.17.1` I see the below issue.

> Note: The tests runs fine when you don't use explicit directory paths like `--coverage-directory` and `--outputFile`

**Issue:**
```
[19:08.9]  @stencil/core
[19:09.1]  v2.17.1 😊
[19:09.3]  testing spec files
[19:09.5]  jest args: --spec --coverage --json --coverage-directory=./dist-coverage/jest-test-coverage
           ./dist-coverage/jest-test-coverage --outputFile=.jest-test-results.json .jest-test-results.json
           --max-workers=8
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
In /Users/mettuman/IdeaProjects/proj-fork
  211 files checked.
  testMatch:  - 0 matches
  testPathIgnorePatterns: /Users/mettuman/IdeaProjects/proj-fork/.vscode,
  /Users/mettuman/IdeaProjects/proj-fork/.stencil, /Users/mettuman/IdeaProjects/proj-fork/node_modules, /Users/mettuman/IdeaProjects/proj-fork/www, /Users/mettuman/IdeaProjects/proj-fork/dist - 71 matches
  testRegex: (/__tests__/.*|(\.|/)(test|spec|e2e))\.[jt]sx?$ - 2 matches
Pattern: ./dist-coverage/jest-test-coverage|.jest-test-results.json - 0 matches
```

**Comment by alicewriteswrongs** [comment]
@manojreddymettu-pro I believe that is a separate issue, see #3481 and #3471

**Comment by alicewriteswrongs** [comment]
@jeripeierSBB to update you, we merged a PR today (#3486) which I believe fixes the underlying issue you're running in to here.

If you want to test it out to make sure it fixes your issue you could:

- clone the repo locally
- do `npm install`, `npm run build` and `npm pack`
- try installing the resulting `.tgz` file in our project by doing `npm install ./path/to/stencil.tgz`

If you'd rather wait for a pre-release or release build to try that out no worries, we will be doing one soon.

Thanks again for reporting this issue!

**Comment by jeripeierSBB** [comment]
> providing some information about the version of yarn you're using, and how yo

@alicewriteswrongs Thanks for all your work on it and sorry for my late response. I'm using yarn v1.22.19, and in my project it really worked with stencil 2.17.0.

I tested your fix and it works with yarn, thank you! Should I close the issue myself?

**Comment by rwaskiewicz** [comment]
The fix for this issue has been officially released as a part of [Stencil v2.17.2](https://github.com/ionic-team/stencil/releases/tag/v2.17.2). As a result, I'm going to close this issue and mark it as completed. If the bug re-appears, please feel free to open a new issue!
